dotCMS/src/main/java/com/dotcms/webdav/FsResource.java

```diff
@@ -0,0 +1,162 @@
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package com.dotcms.webdav;
+
+import com.dotcms.repackage.com.bradmcevoy.http.Auth;
+import com.dotcms.repackage.com.bradmcevoy.http.CollectionResource;
+import com.dotcms.repackage.com.bradmcevoy.http.CopyableResource;
+import com.dotcms.repackage.com.bradmcevoy.http.DigestResource;
+import com.dotcms.repackage.com.bradmcevoy.http.LockInfo;
+import com.dotcms.repackage.com.bradmcevoy.http.LockResult;
+import com.dotcms.repackage.com.bradmcevoy.http.LockTimeout;
+import com.dotcms.repackage.com.bradmcevoy.http.LockToken;
+import com.dotcms.repackage.com.bradmcevoy.http.LockableResource;
+import com.dotcms.repackage.com.bradmcevoy.http.MoveableResource;
+import com.dotcms.repackage.com.bradmcevoy.http.Request;
+import com.dotcms.repackage.com.bradmcevoy.http.Request.Method;
+import com.dotcms.repackage.com.bradmcevoy.http.Resource;
+import com.dotcms.repackage.com.bradmcevoy.http.exceptions.NotAuthorizedException;
+import com.dotcms.repackage.com.bradmcevoy.http.http11.auth.DigestResponse;
+import com.dotmarketing.util.Logger;
+
+
+import java.io.File;
+import java.util.Date;
+
+
+/**
+ *
+ */
+public abstract class FsResource implements Resource, MoveableResource, CopyableResource, LockableResource, DigestResource {
+
+
+    File file;
+    final FileSystemResourceFactory factory;
+    final String host;
+    String ssoPrefix;
+
+    protected abstract void doCopy(File dest);
+
+    public FsResource(String host, FileSystemResourceFactory factory, File file) {
+        this.host = host;
+        this.file = file;
+        this.factory = factory;
+    }
+
+    public File getFile() {
+        return file;
+    }
+
+    public String getUniqueId() {
+        String s = file.lastModified() + "_" + file.length() + "_" + file.getAbsolutePath();
+        return s.hashCode() + "";
+    }
+
+    public String getName() {
+        return file.getName();
+    }
+
+    public Object authenticate(String user, String password) {
+        return factory.getSecurityManager().authenticate(user, password);
+    }
+
+    public Object authenticate(DigestResponse digestRequest) {
+        return factory.getSecurityManager().authenticate(digestRequest);
+    }
+
+    public boolean isDigestAllowed() {
+        return factory.isDigestAllowed();
+    }
+
+    public boolean authorise(Request request, Method method, Auth auth) {
+        boolean b = factory.getSecurityManager().authorise(request, method, auth, this);
+        Logger.info(FileSystemResourceFactory.class,"authorise: result=" + b);
+
+        return b;
+    }
+
+    public String getRealm() {
+        return factory.getRealm(this.host);
+    }
+
+    public Date getModifiedDate() {
+        return new Date(file.lastModified());

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use the Java 8 Date and Time API instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2143)